src/main/java/seedu/address/logic/parser/ExpenseCommandParser.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXPENSE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PARTICIPANT;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.ExpenseCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.activity.Amount;
+
+/**
+ * Parses input arguments and creates a new ExpenseCommand object
+ */
+public class ExpenseCommandParser implements Parser<ExpenseCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ExpenseCommand
+     * and returns an ExpenseCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ExpenseCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_PARTICIPANT, PREFIX_EXPENSE, PREFIX_DESCRIPTION);
+        if (!arePrefixesPresent(argMultimap, PREFIX_PARTICIPANT, PREFIX_EXPENSE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ExpenseCommand.MESSAGE_USAGE));
+        }
+
+        List<String> persons = argMultimap.getAllValues(PREFIX_PARTICIPANT);
```

Comment:
It does get everything in order right? Should add a test some time to ensure this does not change in the future.

Reply:
ok added tests